src/components/UserFilterDropdown/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, {useEffect, useState, ChangeEvent} from 'react';
+import {useDispatch} from 'react-redux';
+import {displayErrorToast} from 'utils/toasts';
+import {AppDispatch, SFC} from 'types';
+import * as S from './Styles';
+import axios from 'axios';
+import {UserReadSerializer} from 'types';
+import {authorizationHeaders} from 'utils/authentication';
+const BASE_URL = `${process.env.REACT_APP_API_URL}/api/users`;
+
+interface User {
+  id: number;
+  username: string;
+  avatar: string;
+}
+const UserFilterDropdown: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [filteredUsers, setFilteredUsers] = useState<User[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const dispatch = useDispatch<AppDispatch>();
+
+  useEffect(() => {
+    // Filter users based on the search term
+    const filterUsers = () => {
+      const filtered = users.filter((user) => user.username.toLowerCase().includes(searchTerm.toLowerCase()));
+      setFilteredUsers(filtered);
+    };
+
+    filterUsers();
+  }, [searchTerm, users]);
+
+  const handleSearch = async (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value);
+    if (event.target.value == '') {
+      setUsers([]);
+      setFilteredUsers([]);
+      return;
+    }
+
+    try {
+      const response = await axios.get<UserReadSerializer, any>(
+        `${BASE_URL}?search=${event.target.value}`,
+        authorizationHeaders(),
+      );
+      setUsers(response.data);
+      setFilteredUsers(response.data);
+    } catch (error) {
+      console.error('Error fetching users:', error);
+    }
+  };
+
+  const handleClick = (user: User) => {
+    setSelectedUser(user);
+  };
+  const removeSelectedUser = () => {
+    setSelectedUser(null);
+  };
+  return (
+    <div>
+      {selectedUser ? (
+        <S.SelectedUserDiv>
+          <S.Table>
+            <tbody>
+              <tr>
+                <td>
+                  <S.Span>{selectedUser.username}</S.Span>
+                </td>
+                <td>
+                  {selectedUser.avatar ? (
+                    <S.Img src={selectedUser.avatar} />
+                  ) : (
+                    <S.Img
+                      src="/static/media/default-avatar.20c02f587c3406ffb40e0e88a890f426.svg"
+                      alt="Default Avatar"
+                    />
```

Comment:
We already have a avatar component that shows the default avatar incase user doesn't have any:
<img width="293" alt="image" src="https://github.com/thenewboston-developers/thenewboston-Frontend/assets/52558124/0defe19b-db0c-4704-bf06-c1fbfcbb2746">

Important: I think, loading/showing anything from `/static/media/` will break in other environments as it's specific to your setup only.